syntax = "proto2";

import "arpc/proto/rpc_extensions.proto";
package multi_call;

option cc_enable_arenas = true;
option cc_generic_services = true;

message MetaNode {
    optional string platform = 1;
    optional string hippo_cluster = 2;
    optional string hippo_app = 3;
    optional string c2_role = 4;
    optional string c2_group = 5;
}

message MetaKV {
    optional string key = 1;
    optional string value = 2;
}

message BizMeta {
    optional string biz_name = 1;
    optional int64 version = 2;
    optional int32 part_count = 3;
    optional int32 part_id = 4;
    optional int32 weight = 5;
    repeated MetaKV extend = 6;
}

message Meta {
    optional MetaNode meta_node = 1;
    repeated BizMeta biz_meta = 2;
}

message HeartbeatRequest {
    repeated BizMeta biz_meta = 1;
}

message HeartbeatResponse {
    optional bool stopped = 1 [default = false];
    optional Meta meta = 2;
}

service HeartbeatService {
    option (arpc.global_service_id) = 777;
    rpc gig_heartbeat(HeartbeatRequest) returns (HeartbeatResponse) {
        option(arpc.local_method_id) = 7;
    };
}
